plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DriverConnectionFactory.java

```diff
@@ -45,24 +47,36 @@ public DriverConnectionFactory(Driver driver, BaseJdbcConfig config, CredentialP
 
     public DriverConnectionFactory(Driver driver, String connectionUrl, Properties connectionProperties, CredentialProvider credentialProvider)
     {
-        this(driver, connectionUrl, connectionProperties, new DefaultCredentialPropertiesProvider(credentialProvider));
+        this(driver, connectionUrl, connectionProperties, new DefaultCredentialPropertiesProvider(credentialProvider), OpenTelemetry.noop());
     }
 
-    public DriverConnectionFactory(Driver driver, String connectionUrl, Properties connectionProperties, CredentialPropertiesProvider<String, String> credentialPropertiesProvider)
+    public DriverConnectionFactory(Driver driver, String connectionUrl, Properties connectionProperties, CredentialProvider credentialProvider, OpenTelemetry openTelemetry)
+    {
+        this(driver, connectionUrl, connectionProperties, new DefaultCredentialPropertiesProvider(credentialProvider), openTelemetry);
+    }
+
+    public DriverConnectionFactory(
+            Driver driver,
+            String connectionUrl,
+            Properties connectionProperties,
+            CredentialPropertiesProvider<String, String> credentialPropertiesProvider,
+            OpenTelemetry openTelemetry)
     {
         this.driver = requireNonNull(driver, "driver is null");
         this.connectionUrl = requireNonNull(connectionUrl, "connectionUrl is null");
         this.connectionProperties = new Properties();
         this.connectionProperties.putAll(requireNonNull(connectionProperties, "connectionProperties is null"));
         this.credentialPropertiesProvider = requireNonNull(credentialPropertiesProvider, "credentialPropertiesProvider is null");
+        this.openTelemetry = requireNonNull(openTelemetry, "openTelemetry is null");
     }
 
     @Override
     public Connection openConnection(ConnectorSession session)
             throws SQLException
     {
         Properties properties = getCredentialProperties(session.getIdentity());
-        Connection connection = driver.connect(connectionUrl, properties);
+        TracingDataSource dataSource = new TracingDataSource(openTelemetry, driver, connectionUrl, properties);
```

Comment:
can we keep dataSource as a field in the class instead of creating it on every `openConnection` call?

Reply:
Moved to a field, but the internal `DataSource` needs to be initialized here either because it depends on `Properties`.